pytest-embedded-idf/pytest_embedded_idf/unity_tester.py

```diff
@@ -422,23 +422,47 @@ def run_all_single_board_cases(
         reset: bool = False,
         timeout: float = 30,
         run_ignore_cases: bool = False,
+        names: t.Optional[list] = None,
+        attributes: t.Optional[dict] = None,
     ) -> None:
         """
         Run all single board cases, including multi_stage cases, and normal cases
 
         Args:
-            group: test case group
-            reset: whether to perform a hardware reset before running a case
-            timeout: timeout. (Default: 30 seconds)
-            run_ignore_cases: run ignored test cases or not
+            group: Test case group. Supports group invert by '!' and groups separated by ','.
```

Comment:
hmm, what will happen when I write '!a,b', it will be explained as which one?

- not in group a, or in group b
- not in group a or b

improve the docstring with `.. note:` would be nice.

Reply:
besides, groups separated by `,` is not a nice idea, since there's no such limitation in our c code. if there's a group named 'a,b', then we got different behavior.

I suggest we support:

```
group: Test case group or a list of test case groups.
```

Reply:
Yes, I agree, this is not a good idea. Let's make them a str or list then.

Reply:
> * not in group a, or in group b
> * not in group a or b

For now it will `not (a and b)`

Reply:
if the logic operand between each list item is "and" it's clear, but counter-intuitive...

For example,

- ['!a', '!b'] is: not in group a, and not in group b
- ['!a', 'b'] is: not in group a, and in group b

if the user want to run group a and group b, they need to call the function twice...

Reply:
Let's do it using 'and' because the unit test for [a][b] will be done using 'and' and not 'or'.
I have created function `validate_group` for it

Reply:
On the other hand, it is easier to just `run_all_single_board_cases` using this function a few times than to create a more complex interface for it.

Reply:
> Let's do it using 'and' because if we compare it to the unit test for [a][b], it will be done using 'and' and not 'or'.

actually on the contrary... unity test case `"test foo" [a][b]` means it could be selected both by `a`, or by `b`.

Reply:
Let's do it like "a & b | c"
WDYT